onnxruntime/core/graph/graph.cc

```diff
@@ -2055,6 +2056,39 @@ void Graph::RemoveInitializedTensor(const std::string& tensor_name) {
   }
 }
 
+Status Graph::ReplaceInitializedTensor(const ONNX_NAMESPACE::TensorProto& new_initializer) {
+  const auto& initializer_name = new_initializer.name();
+  auto name_to_initializer_it = name_to_initial_tensor_.find(initializer_name);
+  ORT_RETURN_IF_NOT(name_to_initializer_it != name_to_initial_tensor_.end(),
+                    "Failed to find existing initializer with name ", initializer_name, ".");
+
+  assert(name_to_initializer_it->second);
+  const auto& old_initializer = *(name_to_initializer_it->second);
+
+  auto dims_eq = [&old_initializer, &new_initializer]() {
+    if (old_initializer.dims_size() != new_initializer.dims_size()) return false;
+    for (int i = 0; i < old_initializer.dims_size(); ++i) {
+      if (old_initializer.dims(i) != new_initializer.dims(i)) return false;
+    }
+    return true;
+  };
+
+  ORT_RETURN_IF_NOT(dims_eq(), "Replacement tensor's dimensions do not match.");
+  ORT_RETURN_IF_NOT(old_initializer.data_type() == new_initializer.data_type(),
+                    "Replacement tensor's data type does not match.");
+
+  auto& mutable_initializers = *graph_proto_->mutable_initializer();
+  auto old_mutable_initializer_ptr_it = std::find(
+      mutable_initializers.pointer_begin(), mutable_initializers.pointer_end(), &old_initializer);
+  assert(old_mutable_initializer_ptr_it != mutable_initializers.pointer_end());
+  assert(*old_mutable_initializer_ptr_it);
```

Comment:
>assert(*old_mutable_initializer_ptr_it); [](start = 2, length = 40)

Not clear what this asssert does. Does it mean that mutable_initializers can have nullptr entries?

Reply:
This check looks redundant since it should be either &old_initializer or mutable_initializers.pointer_end(). Thanks for bringing it up.

Reply:
Regarding whether mutable_initializers can have null entries, I'm not sure.